hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -829,18 +882,50 @@ public void createPool(JobConf conf, PathFilter... filters) {
         super.setMaxSplitSize(minSize);
       }
 
-      InputSplit[] splits = super.getSplits(job, numSplits);
-      ArrayList inputSplitShims = new ArrayList();
+      LOG.info("mapreduce.input.fileinputformat.split.minsize=" + minSize
+          + ", mapreduce.input.fileinputformat.split.maxsize=" + maxSize);
 
-      for (int pos = 0; pos < splits.length; ++pos) {
-        CombineFileSplit split = (CombineFileSplit) splits[pos];
-        if (split.getPaths().length > 0) {
-          inputSplitShims.add(new HadoopShimsSecure.InputSplitShim(job, split.getPaths(), split.getStartOffsets(),
-              split.getLengths(), split.getLocations()));
+      if (isRealTime) {
+        job.set("hudi.hive.realtime", "true");
+        InputSplit[] splits;
+        if (hoodieFilter) {
+          HoodieParquetInputFormat input = new HoodieParquetRealtimeInputFormat();
+          input.setConf(job);
+          splits = input.getSplits(job, numSplits);
```

Comment:
Move this line  splits = input.getSplits(job, numSplits); out of if -else ?

Reply:
Not sure if I get it. the else part calls super.getSplits whereas the if part calls input.getSplits. Let me know if I am missing something.

Reply:
ahh, I missed it, ok thanks